MNE/disp3D/engine/model/3dhelpers/renderable3Dentity.cpp

```diff
@@ -88,56 +88,103 @@ using namespace Qt3DRender;
 
 Renderable3DEntity::Renderable3DEntity(Qt3DCore::QEntity* parent)
 : Qt3DCore::QEntity(parent)
-, m_pCustomMesh(new CustomMesh())
 , m_pTransform(new Qt3DCore::QTransform())
 , m_fRotX(0.0f)
 , m_fRotY(0.0f)
 , m_fRotZ(0.0f)
 {
-    this->addComponent(m_pCustomMesh);
     this->addComponent(m_pTransform);
 }
 
 
 //*************************************************************************************************************
 
-Renderable3DEntity::Renderable3DEntity(const MatrixX3f& tMatVert, const MatrixX3f& tMatNorm, const MatrixX3i& tMatTris, const MatrixX3f& tMatColors, Qt3DCore::QEntity* parent)
-: Qt3DCore::QEntity(parent)
-, m_pCustomMesh(new CustomMesh(tMatVert, tMatNorm, tMatTris, tMatColors))
-, m_pTransform(new Qt3DCore::QTransform())
-, m_fRotX(0.0f)
-, m_fRotY(0.0f)
-, m_fRotZ(0.0f)
+Renderable3DEntity::~Renderable3DEntity()
 {
-    this->addComponent(m_pCustomMesh);
-    this->addComponent(m_pTransform);
+    //releaseNode(this);
 }
 
 
+////*************************************************************************************************************
+
+//void Renderable3DEntity::releaseNode(Qt3DCore::QNode* node)
+//{
+//    if (QEntity* entity = dynamic_cast<QEntity*>(node)) {
+//        QComponentVector components = entity->components();
+
+//        foreach (QComponent* component, components) {
+//            entity->removeComponent(component);
+//            delete component;
+//        }
+//    }
+
+//    QNodeVector nodes = node->childNodes();
+
+//    foreach (QNode* nodeIt, nodes) {
+//        releaseNode(nodeIt);
+//        delete nodeIt;
+//    }
+//}
+
+
 //*************************************************************************************************************
 
-Renderable3DEntity::~Renderable3DEntity()
+void Renderable3DEntity::setTransform(const Qt3DCore::QTransform& transform)
 {
+    if(m_pTransform) {
+        m_pTransform->setMatrix(transform.matrix());
```

Comment:
Agree, if the user changes the transform with no prior inititalized transform pointer, the transform pointer should be created.